dashboard/app/controllers/user_levels_controller.rb

```diff
@@ -58,4 +61,10 @@ def get_level_source
   private def user_level_params
     params.require(:user_level).permit(:best_result, :submitted)
   end
+
+  # Some lab2 levels have a predict_settings field that indicates they are predict levels.
+  # We allow clearing progress on these levels as well as multi or free-response levels.
+  private def clearable_level_type?(level)
+    ['Multi', 'FreeResponse'].include?(level.type) || (level.uses_lab2? && level.predict_settings&.dig("isPredictLevel"))
```

Comment:
nit: is it worth moving `uses_lab2 && predict_settings&.dig("isPredictLevel")` to some helper function on `level.rb` if we're checking this in a bunch of places?

Reply:
I actually did this in a [separate pr](https://github.com/code-dot-org/code-dot-org/pull/59599/files#diff-86af1cb288d4d48f970630d0f21ee11f2dd971239bafe270eafce13691d1d0caR882). Depending on build times I'll either add that in here or do a fast follow to use that function!

Reply:
included it here since drone failed on me 😄